src/library/scanner/libraryscanner.cpp

```diff
@@ -138,7 +138,8 @@ LibraryScanner::LibraryScanner(
     connect(m_pProgressDlg.data(),
             &LibraryScannerDlg::scanCancelled,
             this,
-            &LibraryScanner::slotCancel);
+            &LibraryScanner::slotCancel,
+            Qt::DirectConnection);
```

Comment:
Making this a direct connection introduces some concurrency issues. Is this strictly required for the solution?

Reply:
Taking a look at `slotCancel()` -> `cancel()`, the latter does `m_pool.waitForDone()`.
Isn't that sufficient?